### PR TITLE
Feature Added: AddTodo Component

### PR DESCRIPTION
**Pull request by ksrscharan**
## Description
- Add Todo Component Added

## Database schema changes
NA

### Automated test cases added
NA

### Manual test cases run
NA

### CODE REVIEW

**Comment by ksrscharan** [comment]
Screenshot 2022-06-27 at 1 33 04 PM

File: src/components/AddTodo.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React, { useContext } from "react";
+import { ACTION } from "../context/Actions";
+import { GlobalContext } from "../context/GlobalState";
+
+
+
+function AddTodo() {
+  const { state, dispatch } = useContext(GlobalContext); //type here
```
Comment (sourabh-4097) [inline src/components/AddTodo.tsx]: remove comment

File: src/components/AddTodo.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React, { useContext } from "react";
+import { ACTION } from "../context/Actions";
+import { GlobalContext } from "../context/GlobalState";
+
+
```
Comment (sourabh-4097) [inline src/components/AddTodo.tsx]: remove extra lines

File: src/context/Actions.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+export const ACTION: ActionType = {
+    REGISTER_TODO: "register-todo",
+    ADD_TODO: "add-todo",
+    DELETE_TODO: "delete-todo",
+    EDIT_TODO: "edit-todo",
+    REGISTER_UPDATE: "register-update",
+    UPDATE_TITLE: "update-name",
+    CANCEL_UPDATE: "cancel-update",
+    REGISTER_DATE: "register-date",
+  };
```
Comment (sourabh-4097) [inline src/context/Actions.tsx]: add new line

File: src/context/GlobalState.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import React, {
+  createContext,
+  ReactNode,
+  useReducer,
+} from "react";
+import Reducer from "./Reducer";
+
+
```
Comment (sourabh-4097) [inline src/context/GlobalState.tsx]: extra line remove

File: src/context/Actions.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+export type ActionType = {
+  REGISTER_TODO: string;
```
Comment (sourabh-4097) [inline src/context/Actions.tsx]: arrange alphabetically

File: src/context/Actions.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  ADD_TODO: string;
+  DELETE_TODO: string;
+  EDIT_TODO: string;
+  REGISTER_UPDATE: string;
+  UPDATE_TITLE: string;
+  CANCEL_UPDATE: string;
+  REGISTER_DATE: string;
+}
+export const ACTION: ActionType = {
+    REGISTER_TODO: "register-todo",
```
Comment (sourabh-4097) [inline src/context/Actions.tsx]: alphabetical

File: src/context/GlobalState.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+interface StateType{
+  todo: {}[] | {id: number; title: string; date: string}[];
```
Comment (sourabh-4097) [inline src/context/GlobalState.tsx]: alphabetical

File: src/context/Reducer.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
+import { ACTION } from "./Actions";
+
+export default function reducer(state: any, action: any) { //type here
```
Comment (sourabh-4097) [inline src/context/Reducer.tsx]: remove comment

**Review by sourabh-4097 — CHANGES_REQUESTED** [review]

File: src/context/GlobalState.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+export const GlobalContext = createContext(initialState);
+
+export function GlobalProvider({ children }: { children: ReactNode }) {
+  const [state, dispatch] = useReducer(Reducer, initialState);
+
+  return (
+    <GlobalContext.Provider value={{ state, dispatch }}>
+      {" "}
+      {/* type here */}
```
Comment (sourabh-4097) [inline src/context/GlobalState.tsx]: remove comment

File: src/context/GlobalState.tsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  addTodo: string;
+  date: string;
+  isEditing: boolean;
+  selectedEdit: {} | { id: number; date: string; title: string };
+  todo: {}[] | { id: number; date: string; title: string }[];
+  updatedTitle: string;
+}
+
+const initialState: StateType | any = {
+  todo: [],
```
Comment (sourabh-4097) [inline src/context/GlobalState.tsx]: alphabetical

**Pull request merged**